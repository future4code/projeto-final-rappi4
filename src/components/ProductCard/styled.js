import styled from 'styled-components'

export const StyledCardContainer = styled.div`
    height: 112.6px;
    border-radius: 8px;
    border: solid 1px #b8b8b8;
    display: flex;
    position: relative;
`

export const StyledImg = styled.img`
    width: 97px;
    height: 111px;
    border-radius: 8px 0 0 8px;
    background-color: #d8d8d8;
`

export const StyledCardContent = styled.div`
    padding: 16px;
`

export const StyledTitleSpan = styled.span`
    display: block;
    color: #e86e5a;
    font-size: 16px;
    margin-bottom: 6px;
`
export const StyledContentSpan = styled.span`
    display: block;
    font-size: 14px;
    color: #b8b8b8;
    margin-bottom: 6px;
`
export const StyledPriceSpan = styled.span`
    display: block;
    font-size: 16px;
`
export const StyledButton = styled.button`
    position: absolute;
    bottom: 0;
    right: 0;
    width: 90px;
    height: 31px;
    border: 1px solid black;
    border-radius: 8px 0 8px 0;
    background: white;
    color: black;
    font-size: 12px;
    white-space: nowrap;
    text-decoration: none;
    padding: 0.25rem 0.5rem;
    cursor: pointer;
`

export const StyledCounter = styled.span`
    position: absolute;
    top: 0;
    right: 0;
    width: 33px;
    height: 33px;
    border: 1px solid #e86e5a;
    border-radius: 0 8px 0 8px;
    background: white;
    color: #e86e5a;
    font-size: 16px;
    display: flex;
    justify-content: center;
    align-items: center;
`