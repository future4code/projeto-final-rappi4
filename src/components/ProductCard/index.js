import React from 'react'
import { StyledCardContainer, StyledImg, StyledCardContent, StyledTitleSpan, StyledContentSpan, StyledPriceSpan, StyledButton, StyledCounter } from './styled';


export default function ProductCard(props) {

    return (
        <StyledCardContainer>
            <StyledImg src={props.product.photoUrl} />
            <StyledCardContent>
                <StyledTitleSpan>{props.product.name}</StyledTitleSpan>
                <StyledContentSpan>{props.product.description}</StyledContentSpan>
                <StyledPriceSpan>R$ 0{props.product.price},00</StyledPriceSpan>
            </StyledCardContent>
            <StyledCounter>2</StyledCounter>
            <StyledButton>adicionar</StyledButton>
        </StyledCardContainer>
    )
}
