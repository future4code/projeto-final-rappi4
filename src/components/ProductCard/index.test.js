import React from 'react';
import renderer from 'react-test-renderer';
import ProductCard from './index';



describe("Product Card",()=>{
    it("should render ",()=>{
        const mockCard = {
            "id": "CnKdjU6CyKakQDGHzNln",
            "category": "Salgado",
            "price": "1",
            "photoUrl": "https://static-images.ifood.com.br/image/upload/f_auto,t_high/pratos/65c38aa8-b094-413d-9a80-ddc256bfcc78/201907031404_66194495.jpg",
            "name": "Bibsfiha carne",
            "description": "Esfiha deliciosa, receita secreta do Habibs."
        }

        const tree = renderer.create(<ProductCard product={mockCard}/>).toJSON();

        expect(tree).toMatchSnapshot();
    })
})