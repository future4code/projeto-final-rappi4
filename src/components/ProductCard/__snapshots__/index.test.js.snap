// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Card should render  1`] = `
<div
  className="sc-bdVaJa jkBsrb"
>
  <img
    className="sc-bwzfXH hhIhSy"
    src="https://static-images.ifood.com.br/image/upload/f_auto,t_high/pratos/65c38aa8-b094-413d-9a80-ddc256bfcc78/201907031404_66194495.jpg"
  />
  <div
    className="sc-htpNat eYSXSN"
  >
    <span
      className="sc-bxivhb qUNyQ"
    >
      Bibsfiha carne
    </span>
    <span
      className="sc-ifAKCX gwVnnc"
    >
      Esfiha deliciosa, receita secreta do Habibs.
    </span>
    <span
      className="sc-EHOje frzsaL"
    >
      R$ 0
      1
      ,00
    </span>
  </div>
  <span
    className="sc-gzVnrw kFsFDa"
  >
    2
  </span>
  <button
    className="sc-bZQynM jMfwPM"
  >
    adicionar
  </button>
</div>
`;
